src/logic/safe/transactions/gas.js

```diff
@@ -67,6 +68,8 @@ export const generateTxGasEstimateFrom = async (
 ) => {
   try {
     let safeInstance = safe
+    const ready = await onboardUser()
```

Comment:
could you please explain why you included `onboardUser` to this function?
Feels weird to me, because:
1. We don't have actions to estimate gas for if the user hasn't connected a wallet
2. You call a function to estimate gas, and it runs a side effect trying to get user to connect a wallet

Reply:
According to what Aaron from BlockNative said, I thought a check before every web3 method were needed. But you are rigth that in this case, one check before sending a transaction is enough.